rest/aggregates/aggregates.go

```diff
@@ -0,0 +1,45 @@
+package aggregates
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/polygon-io/client-golang/rest/client"
+)
+
+// todo: add comments for godoc
+
+const (
+	GetPath               = "/v2/aggs/ticker/{ticker}/range/{multiplier}/{resolution}/{from}/{to}"
+	GetPreviousClosePath  = "/v2/aggs/ticker/{ticker}/prev"
+	GetGroupedDailyPath   = "/v2/aggs/grouped/locale/{locale}/market/{marketType}/{date}"
+	GetDailyOpenClosePath = "/v1/open-close/{ticker}/{date}"
+)
+
+type Client struct {
+	client.HTTPBase
+}
+
+func (ac *Client) Get(ctx context.Context, params GetParams, opts ...client.Option) (*AggsResponse, error) {
+	res := &AggsResponse{}
+	err := ac.Call(http.MethodGet, GetPath, params, res, append([]client.Option{client.WithContext(ctx)}, opts...)...)
+	return res, err
+}
+
+func (ac *Client) GetPreviousClose(ctx context.Context, params GetPreviousCloseParams, opts ...client.Option) (*AggsResponse, error) {
```

Comment:
This is definitely up for debate. We ended up not following our current convention here (path params as inputs and query params as a struct ptr) for a couple reasons:
- some of our endpoints have a lot of path params; long method signatures seem harder to use and document
- using a type like this allows us to avoid calling `Sprintf` on our endpoint paths and we can set path and query params with resty instead

Reply:
Oops I think I was looking at an older commit. I like what you've done with the query param struct within the params struct